oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticIndexStatistics.java

```diff
@@ -164,77 +157,59 @@ static <K, V> LoadingCache<K, V> setupCache(long maxSize, long expireMin, long r
             } else {
-                countRequest.query(QueryBuilders.matchAllQuery());
+                cBuilder.query(q -> q.matchAll( m-> m));
             }
-
-            CountResponse response = crd.connection.getClient().count(countRequest, RequestOptions.DEFAULT);
-            return (int) response.getCount();
+            return (int) crd.connection.getClient().count(cBuilder.build()).count();
         }
     }
 
     static class StatsCacheLoader extends CacheLoader<StatsRequestDescriptor, StatsResponse> {
 
-        private static final ObjectMapper MAPPER = new ObjectMapper();
-
         @Override
-        public StatsResponse load(StatsRequestDescriptor countRequestDescriptor) throws IOException {
+        public StatsResponse load(@NotNull StatsRequestDescriptor countRequestDescriptor) throws IOException {
             return stats(countRequestDescriptor);
         }
 
         @Override
-        public ListenableFuture<StatsResponse> reload(StatsRequestDescriptor crd, StatsResponse oldValue) {
+        public ListenableFuture<StatsResponse> reload(@NotNull StatsRequestDescriptor crd, @NotNull StatsResponse oldValue) {
             ListenableFutureTask<StatsResponse> task = ListenableFutureTask.create(() -> stats(crd));
             Executors.newSingleThreadExecutor().execute(task);
             return task;
         }
 
         private StatsResponse stats(StatsRequestDescriptor crd) throws IOException {
-            RestClient lowLevelClient = crd.connection.getClient().getLowLevelClient();
-            // TODO: the elastic rest high-level client does not currently support the index stats API.
-            // We should switch to this once available, since the _cat API is not intended for use by applications
-            Response response = lowLevelClient.performRequest(
-                    new Request("GET", "/_cat/indices/" + crd.index
-                            + "?format=json&h=store.size,creation.date,docs.count,docs.deleted&bytes=b&time=ms")
-            );
-
-            if (response != null) {
-                String rawBody = EntityUtils.toString(response.getEntity());
-                TypeReference<List<Map<String, String>>> typeRef = new TypeReference<List<Map<String, String>>>() {};
-                List<Map<String, String>> indices = MAPPER.readValue(rawBody, typeRef);
-
-                if (!indices.isEmpty()) {
-                    // we ask for a specific index, so we can get the first entry
-                    Map<String, String> indexProps = indices.get(0);
-                    String size = indexProps.get("store.size");
-                    String creationDate = indexProps.get("creation.date");
-                    String luceneDocsCount = indexProps.get("docs.count");
-                    String luceneDocsDeleted = indexProps.get("docs.deleted");
-                    return new StatsResponse(
+            IndicesRecord record = crd.connection.getClient().cat().indices(i -> i
```

Comment:
Could this be null (similar to the response earlier on ?)
. If yes, then we could possibly have a potential NPE

Reply:
According to the documentation, nulls are prevented by the new library. I will double-check just in case.